ibm/service/power/resource_ibm_pi_shared_processor_pool.go

```diff
@@ -380,11 +382,35 @@ func resourceIBMPISharedProcessorPoolUpdate(ctx context.Context, d *schema.Resou
 		return diag.Errorf("error updating the shared processor pool: %v", err)
 	}
 
-	if d.HasChange(Attr_SharedProcessorPoolPlacementGroups) {
+	diagErr := detectSPPPlacementGroupChange(ctx, sess, cloudInstanceID, d, sppID)
+	if diagErr != nil {
+		return diagErr
+	}
 
-		pgClient := st.NewIBMPISPPPlacementGroupClient(ctx, sess, cloudInstanceID)
+	if d.HasChange(Arg_UserTags) {
+		if crn, ok := d.GetOk(Attr_CRN); ok {
+			oldList, newList := d.GetChange(Arg_UserTags)
+			err := flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, crn.(string), "", UserTagType)
+			if err != nil {
+				log.Printf("Error on update of pi shared processor pool (%s) pi_user_tags: %s", sppID, err)
+			}
+		}
+	}
+
+	return resourceIBMPISharedProcessorPoolRead(ctx, d, meta)
+}
 
-		oldRaw, newRaw := d.GetChange(Attr_SharedProcessorPoolPlacementGroups)
+func detectSPPPlacementGroupChange(ctx context.Context, sess *ibmpisession.IBMPISession, cloudInstanceID string, d *schema.ResourceData, sppID string) diag.Diagnostics {
+	if d.HasChange(Arg_SharedProcessorPoolPlacementGroups) || d.HasChange(Attr_SharedProcessorPoolPlacementGroups) {
```

Comment:
```suggestion
	if HasChanges(Arg_SharedProcessorPoolPlacementGroups, Attr_SharedProcessorPoolPlacementGroups) {
```
This be group together

Reply:
updated